frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1203,7 +1215,10 @@ func dispatchCopy(d *dispatchState, cfg copyConfig) error {
 		commitMessage.WriteString(" <<" + src.Path)
 
 		data := src.Data
-		f := src.Path
+		f, err := system.CheckSystemDriveAndRemoveDriveLetter(src.Path)
```

Comment:
Instead of propagating this call throughout `dispatchCopy`, WDYT about pushing it down into `llb.Copy` (or `fileActionCopy`)? I'm not sure, I'm not as familiar with that code, but to me it seems like then we'd avoid having to copy this block everywhere, and other frontends would also automatically get the path fixes for Windows?

Reply:
There are a couple of places where the source folder passed to ```llb.Copy()``` is wrapped in ```filepath.Join("/")```: 

https://github.com/moby/buildkit/blob/6d9c24dc3aed1b98cc7697016bf6d6d5e5e63b70/frontend/dockerfile/dockerfile2llb/convert.go#L1148-L1153

If this path is a Windows path, this ends up becoming something like: ```\C:\test```. If the requirement is to make sure the path is absolute before it gets passed to ```llb.Copy()``` here, the assumption being that it is possible this path may be relative in other places, we need to make sure the path is properly converted before we pass it in. Maybe a new function called ```ToAbolute(path string, platform string)```, that does the right thing based on target platform? What do you think?

Reply:
friendly ping